lib/turbograft/xhr_headers.rb

```diff
@@ -25,23 +25,31 @@ def _compute_redirect_to_location(*args)
     end
 
     private
-      def store_for_turbolinks(url)
-        session[:_turbolinks_redirect_to] = url if session && request.headers["X-XHR-Referer"]
-        url
-      end
+    # Ensure backwards compatibility
+    # Rails < 4.2:  _compute_redirect_to_location(options)
+    # Rails >= 4.2: _compute_redirect_to_location(request, options)
+    def _normalize_redirect_params(args)
+      options, req = args.reverse
+      [options, req || request]
+    end
 
-      def set_xhr_redirected_to
-        if session && session[:_turbolinks_redirect_to]
-          response.headers['X-XHR-Redirected-To'] = session.delete :_turbolinks_redirect_to
-        end
-      end
+    def store_for_turbolinks(url)
+      session[:_turbolinks_redirect_to] = url if session && request.headers["X-XHR-Referer"]
+      url
+    end
 
-      # Ensure backwards compatibility
-      # Rails < 4.2:  _compute_redirect_to_location(options)
-      # Rails >= 4.2: _compute_redirect_to_location(request, options)
-      def _normalize_redirect_params(args)
-        options, req = args.reverse
-        [options, req || request]
+    def set_xhr_redirected_to
+      if redirect_was_sent? && request_matches_redirect?
```

Comment:
I was a little confused by whether the redirect was being sent or received – what about renaming it to `redirected?`, `xhr_redirected?`, or `turbolinks_redirected?`